dm/worker/server.go

```diff
@@ -138,19 +142,23 @@ func (s *Server) Start() error {
 	}()
 
 	s.wg.Add(1)
-	go func() {
-		defer s.wg.Done()
-		// TODO: handle fatal error from observeSourceBound
-		//nolint:errcheck
-		s.observeSourceBound(s.ctx, revBound)
-	}()
+	s.startKeepAlive()
 
 	s.wg.Add(1)
 	go func() {
 		defer s.wg.Done()
-		// worker keepalive with master
-		// If worker loses connect from master, it would stop all task and try to connect master again.
-		s.KeepAlive()
+		for {
+			//nolint:errcheck
+			s.observeSourceBound(s.ctx, revBound)
+			s.stopKeepAlive()
+			select {
+			case <-s.ctx.Done():
+				return
+			case <-time.After(keepaliveTimeout):
```

Comment:
sleep time should be related to `s.cfg.KeepAliveTTL`, for example, `2 * s.cfg.KeepAliveTTL`